targetlibs/nrf5x/components/nfc/t2t_lib/hal_t2t/hal_nfc_t2t.h

```diff
@@ -122,9 +158,8 @@ ret_code_t hal_nfc_start(void);
 
 /** @brief Function for sending a packet to the connected NFC reader.
   *
-  * The provided data buffer belongs to the caller and is guaranteed to be
-  * valid until the HAL_NFC_EVENT_DATA_TRANSMITTED event is received by the
-  * callback.
+  * The provided data buffer belongs to the caller and may be freed after this

```

Comment:
It shouldn't really lead to fragmentaton, although allocating actual flat buffers can take quite a while. I guess one option might be to create a 255 byte JsVar-based buffer on `nfcStart` - so if NFC isn't being used, that data is available for other stuff.

Reply:
I will look into that. Assuming rx will also be stored in this JVar. Could `NFCrx` be called with a view instead of a copy. In turn, speeding up request processing?

Reply:
That sounds like a great idea! Allocating an ArrayBufferView would be super quick (it's one var), and actually would allow you to use the new queue-based stuff I added recently (you'd just push the length of received data into the queue along with BLEP_NFCRX). 

The only gotcha is the variable's contents wouldn't be guaranteed to be correct if used after event itself, but that's not like it'd be a segfault or anything.